core/utils/poly-look/src/incubate/visualisations/BubbleCluster.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef, useEffect } from "react";
+import * as d3 from "d3";
+
+export const BubbleCluster = ({
+  data,
+  width,
+  height,
+  bubbleColor = "blue",
+  textColor = "white",
+  opacity = 1,
+  showValues = true,
+  onBubbleClick = () => {},
+}) => {
+  const bubbleRef = useRef();
+  const edgePadding = 5;
+  const smallBubblesRadius = 20;
+  const bigBubblesRadius = 50;
+  const bigBubblesFont = "20px";
+  const mediumBubblesFont = "16px";
+
+  function makeHierarchy(children) {
+    return d3.hierarchy({ children }).sum((d) => d.value);
+  }
+
+  function pack() {
+    return d3
+      .pack()
+      .size([width - edgePadding, height - edgePadding])
+      .padding(3);
+  }
+
+  function createSvg() {
+    return d3
+      .select(bubbleRef.current)
+      .append("svg")
+      .attr("viewBox", `0 0 ${width} ${height}`);
+  }
+
+  function drawClusteredBubbles(svg) {
```

Comment:
I think it's good enough for a first version, but as all those comments suggest, I think this is a nice candidate for splitting up the logic into several smaller, more atomic functions. I _think_ I did that relatively nicely in `CompanyBubbles` over in polyExplorer, but I may be wrong :D

Anyway, I'd leave that for after this PR lands.

Reply:
Created a follow up ticket for your feedback here https://jira.polypoly.eu/browse/PROD4POD-868